app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,91 @@
+class OrdersController < ApplicationController
+  
+  before_action :find_user, only:[:complete_purchase, :purchase_confirmation]
+  before_action :find_cart, only:[:complete_purchase, :purchase_confirmation, :add]
+
+  def show
+    #assert logged in
+    @order = Order.find_by(id: params[:id])
+    @user = @order&.user
+  end
+  
+  def complete_purchase
+    if @cart.order_items.count == 0
+      flash[:status] = :failure
+      flash[:result_text] = "Must have an item in cart to complete purchase"
+      redirect_to cart_path
+      return
+    end
+    
+    if @cart.status = "pending"
+      verification = User.verify_user_at_purchase(@cart.user)
+      if verification != nil
+        #there are validation errors from the user model
+        flash[:status] = :failure
+        flash[:messages] = verification
+        redirect_to edit_user_path(@cart.user.id)
+        return
+      end
+    end
+    
+    purchase_confirmation
+  end
+  
+  def purchase_confirmation
+    @cart.status = "paid"
+    
+    @cart.date_time_order_purchased = DateTime.now
+    
+    @cart.date_time_order_purchased
```

Comment:
This line doesn't do anything. This line queries for `@cart`'s attribute of `date_time_order_purchased`, but doesn't actually execute any commands or do anything. I want to make sure that your team gets some understanding about why this line doesn't do anything soon; come see me!